### PR TITLE
v0.4.0

### PR DESCRIPTION
**Pull request by stuft2**
- BREAKING CHANGE: Drop node 12 support
- BREAKING CHANGE: Drop support for Pino v6
- chore(deps): update dependencies
- chore(license): update year
- feat: export ByuLogger
- fix: improve type check
- style: lint test

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,45 +1,51 @@
 {
   "name": "@byu-oit/logger",
-  "version": "0.3.14",
+  "version": "0.5.0",
```
Comment (stuft2) [inline package.json:3]: Looks like version 0.4.0 was never deployed 👀

File: src/logger.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
-import Pino from 'pino'
-import { getLevel, isInstalled } from './util'
+import { Logger, LoggerOptions, pino } from 'pino'
```
Comment (stuft2) [inline src/logger.ts:1]: This is preferred over default exports. I was seeing the same error as in this issue: https://github.com/pinojs/pino/issues/1452#issuecomment-1153335237

File: package.json (hunk 1 above)
Comment (stuft2) [reply package.json:3]: We could go back and fix the history, but it seems like extra effort at this point.

File: src/logger.ts [hunk 3]
```diff
@@ -16,11 +16,18 @@ export default function DefaultLogger (options?: Pino.LoggerOptions): Pino.Logge
       censor: '***'
     },
     // if in local environment use pretty print logs
-    ...process.env.NODE_ENV !== 'production' && isInstalled('pino-pretty') && {
-      prettyPrint: { translateTime: 'UTC:yyyy-mm-dd\'T\'HH:MM:ss.l\'Z\'' } // show timestamp instead of epoch time
+    // ...process.env.NODE_ENV !== 'production' && isInstalled('pino-pretty') && {
+    //   prettyPrint: { translateTime: 'UTC:yyyy-mm-dd\'T\'HH:MM:ss.l\'Z\'' } // show timestamp instead of epoch time
```
Comment (yoshutch) [inline src/logger.ts:20]: (nit) do we need these commented lines commented anymore?

Comment (stuft2) [reply src/logger.ts:20]: No, we can remove those 👍🏻

**Review by yoshutch — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (stuft2) [reply package.json]: The version jump bothered me, so I fixed it

**Review by yoshutch — APPROVED** [review]

**Review by yoshutch — APPROVED** [review]

**Pull request merged**